pkg/yurthub/proxy/local/faketoken.go

```diff
@@ -0,0 +1,112 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package local
+
+import (
+	"bytes"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/klog/v2"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/kubernetes/serializer"
+	"github.com/openyurtio/openyurt/pkg/yurthub/util"
+)
+
+func WithFakeTokenInject(handler http.Handler, serializerManager *serializer.SerializerManager) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		info, _ := apirequest.RequestInfoFrom(ctx)
+		if info.Resource == "serviceaccounts" && info.Subresource == "token" {
+			klog.Infof("Find serviceaccounts token request, write fake token to response.")
+			var buf bytes.Buffer
+			headerNStr := req.Header.Get("Content-Length")
+			headerN, _ := strconv.Atoi(headerNStr)
+			n, err := buf.ReadFrom(req.Body)
+			if err != nil || (headerN != 0 && int(n) != headerN) {
+				klog.Warningf("read body of post request when cluster is unhealthy, expect %d bytes but get %d bytes with error, %v", headerN, n, err)
+			}
+			tokenRequset, err := getTokenRequestWithFakeToken(buf.Bytes(), req, info, serializerManager)
+			if err != nil {
+				klog.Errorf("failed to inject fake token for request %s when cluster is unhealthy with error, %v", util.ReqString(req), err)
+				copyHeader(w.Header(), req.Header)
+				w.WriteHeader(http.StatusCreated)
+				nw, err := w.Write(buf.Bytes())
+				if err != nil || nw != int(n) {
+					klog.Errorf("write resp for token request when cluster is unhealthy, expect %d bytes but write %d bytes with error, %v", n, nw, err)
+				}
+			} else {
+				klog.Infof("Write fake token for request %v", util.ReqString(req))
+				err := util.WriteObject(http.StatusCreated, tokenRequset, w, req)
+				if err != nil {
+					klog.Errorf("write fake token resp for token request when cluster is unhealthy with error, %v", err)
+				}
+			}
+
+		} else {
+			handler.ServeHTTP(w, req)
+		}
+	})
+}
+
+func getTokenRequestWithFakeToken(data []byte, req *http.Request, info *apirequest.RequestInfo, serializerManager *serializer.SerializerManager) (*authv1.TokenRequest, error) {
+	ctx := req.Context()
+	reqContentType, _ := util.ReqContentTypeFrom(ctx)
+	respContentType := reqContentType
+	serializer := serializerManager.CreateSerializer(respContentType, info.APIGroup, info.APIVersion, info.Resource)
```

Comment:
gvr from request maybe "/v1/serviceaccount", but `TokenRequest` will be decoded here, so we need to use `gvr=authentication.k8s.io/v1/tokenrequest` to create `serializer`.

Reply:
moreover, we need to create only one serializer for decoding `gvr=authentication.k8s.io/v1/tokenrequest`

Reply:
> moreover, we need to create only one serializer for decoding `gvr=authentication.k8s.io/v1/tokenrequest`

SerializerManager will cache yurtClientNegotiator if create serializer succeed. If we create serializer for gvr=authentication.k8s.io/v1/tokenrequest repeatedly, it will create serializer from the cache. So, does it make no sense to create only one serializer?
```
if cn, ok := sm.ClientNegotiators[gvr]; ok {
    clientNegotiator, recognized = cn.ClientNegotiator, cn.recognized
} else {
    clientNegotiator, recognized = sm.genClientNegotiator(gvr)
    sm.ClientNegotiators[gvr] = &yurtClientNegotiator{
	recognized:       recognized,
	ClientNegotiator: clientNegotiator,
    }
}
```

Reply:
@LinFCai Yes, you are right, there is no need to create serializer again. but please use `gvr=authentication.k8s.io/v1/tokenrequest` to create serializer.

Reply:
OK， now I have used gvr=authentication.k8s.io/v1/tokenrequest to create serializer.